### PR TITLE
Add option to specify custom class names for OpenAPI and AsyncAPI schemas

### PR DESCRIPTION
**Pull request by dkarwacki**

### CODE REVIEW

File: docs/asyncapi-docs/src/main/scala/sttp/tapir/docs/asyncapi/AsyncAPIDocsOptions.scala [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 
 import sttp.tapir.Endpoint
+import sttp.tapir.SchemaType.SObjectInfo
 
 case class AsyncAPIDocsOptions(
-    subscribeOperationId: (Vector[String], Endpoint[_, _, _, _]) => String,
-    publishOperationId: (Vector[String], Endpoint[_, _, _, _]) => String
+                                subscribeOperationId: (Vector[String], Endpoint[_, _, _, _]) => String,
+                                publishOperationId: (Vector[String], Endpoint[_, _, _, _]) => String,
+                                schemaObjectInfoToNameMapper: SObjectInfo => String = SObjectInfo.defaultToNameMapper
```
Comment (adamw) [inline docs/asyncapi-docs/src/main/scala/sttp/tapir/docs/asyncapi/AsyncAPIDocsOptions.scala:9]: maybe let's use a simpler name - I think just `schemaName` would be sufficient, the rest is obvious from the type signature

File: core/src/main/scala/sttp/tapir/SchemaType.scala [hunk 2]
```diff
@@ -73,6 +73,10 @@ object SchemaType {
   case class SObjectInfo(fullName: String, typeParameterShortNames: List[String] = Nil)
   object SObjectInfo {
     val Unit: SObjectInfo = SObjectInfo(fullName = "Unit")
+    val defaultToNameMapper: SObjectInfo => String = info => {
```
Comment (adamw) [inline core/src/main/scala/sttp/tapir/SchemaType.scala:76]: this is only used for openapi/asyncapi generation, so I'd keep that code for use only in these modules. Maybe the package object `sttp.tapir.docs.apispec` would be a good place?

File: docs/asyncapi-docs/src/main/scala/sttp/tapir/docs/asyncapi/AsyncAPIDocsOptions.scala (hunk 1 above)
Comment (dkarwacki) [reply docs/asyncapi-docs/src/main/scala/sttp/tapir/docs/asyncapi/AsyncAPIDocsOptions.scala]: corrected

File: core/src/main/scala/sttp/tapir/SchemaType.scala (hunk 2 above)
Comment (dkarwacki) [reply core/src/main/scala/sttp/tapir/SchemaType.scala]: corrected

**Comment by adamw** [comment]
Thank you!

**Pull request merged**